src/meta/expr.lean

```diff
@@ -332,7 +332,9 @@ meta def contains_constant (e : expr) (p : name → Prop) [decidable_pred p] : b
 e.fold ff (λ e' _ b, if p (e'.const_name) then tt else b)
 
 /-- `get_simp_args e` returns the arguments of `e` that simp can reach via congruence lemmas. -/
-meta def get_simp_args (e : expr) : tactic (list expr) := do
+meta def get_simp_args (e : expr) : tactic (list expr) :=
+-- `mk_specialized_congr_lemma_simp` throws an assertion violation if its argument is not an app
```

Comment:
Did this change in 3.9.0? I don't remember any related PRs. (Either way, I don't mind if this fix is included.)

Reply:
I don't think so, but for testing I've built mathlib with a debug build of Lean.  (The assertion violation does not happen in release versions.)